src/main/java/ch/njol/skript/effects/EffKnockback.java

```diff
@@ -0,0 +1,93 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package ch.njol.skript.effects;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.RequiredPlugins;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Effect;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser;
+import ch.njol.skript.util.Direction;
+import ch.njol.util.Kleenean;
+import org.bukkit.entity.LivingEntity;
+import org.bukkit.event.Event;
+import org.bukkit.util.Vector;
+import org.eclipse.jdt.annotation.Nullable;
+
+
+@Name("Knockback")
+@Description("Knocks a living entity in a direction. Mechanics such as knockback resistance will be factored in.")
+@Examples({"knockback player north",
+	"knock victim (vector from attacker to victim) with strength 10"})
+@Since("INSERT VERSION")
+@RequiredPlugins("Paper 1.19.2+")
+public class EffKnockback extends Effect {
+
+	static {
+		if (Skript.methodExists(LivingEntity.class, "knockback", double.class, double.class, double.class))
+			Skript.registerEffect(EffKnockback.class, "knock[back] %livingentities% %direction% [with (strength|force) %-number%]");
+	}
+
+	@SuppressWarnings("null")
+	private Expression<LivingEntity> entityExpr;
+	@SuppressWarnings("null")
+	private Expression<Direction> directionExpr;
+	@Nullable
+	private Expression<Number> strengthExpr;
+
+	@SuppressWarnings({"unchecked", "null"})
+	@Override
+	public boolean init(final Expression<?>[] exprs, final int matchedPattern, final Kleenean isDelayed, final SkriptParser.ParseResult parseResult) {
```

Comment:
```suggestion
	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, final ParseResult parseResult) {
```
Imports need to be adjusted.